pykrige/rk.py

```diff
@@ -78,30 +96,33 @@ def predict(self, x, *args, **kwargs):
         if not self.model:
             raise Exception('Not trained. Train first')
 
-        return self.execute(x, *args, **kwargs)[0]
+        points = self._dimensionality_check(x)
 
-    def execute(self, x, *args, **kwargs):
+        return self.execute(points, *args, **kwargs)[0]
+
+    def execute(self, points, *args, **kwargs):
+        # TODO array of Points, (x, y) pairs of shape (N, 2)
         """
         Parameters
         ----------
-        x: ndarray
-            array of Points, (x, y) pairs of shape (N, 2)
+        points: dict
 
         Returns:
         -------
         Prediction array
         Variance array
         """
-        if isinstance(self.model, OrdinaryKriging):
+        if isinstance(self.model, OrdinaryKriging) or \
+                isinstance(self.model, OrdinaryKriging3D):
             prediction, variance = \
-                self.model.execute('points', x[:, 0], x[:, 1],
+                self.model.execute('points',
                                    n_closest_points=self.n_closest_points,
-                                   backend='loop')
+                                   backend='loop',
+                                   **points)
```

Comment:
@rth 
Yes, it's kinda inconvenient that the original had `val` and `z` has been mixed up. And also (x, y, z) and (xpoints, y...).

Anyway I have addressed backward compatibility, but that required some patch job.

Thank you so much as always for your excellent feedback :+1: